### ISSUE TITLE
This image abnormally uses the CPU after 3600 seconds.

### ISSUE THREAD

**Issue by AKKYM** [buchdag/multiarch-letsencrypt-nginx-proxy#1]
The process called letsencryt-service is abnormally using CPU.
After rebooting 3600 s is working normally, but after that the CPU has been used abnormally.
I am using it with armhf. (Scaleway C1)
What is the problem?

Are you mining anyway?

**Comment by buchdag** [comment]
> Are you mining anyway?

What do you mean ?

I did not do any benchmark beside obtaining certificates from both staging and production Let's Encrypt API endpoint on the two scaleway arm VM and a Raspberry Pi 3. This repo is more a proof of concept than anything production ready.

`armhf` is super slow with pretty much anything crypto related, specifically the 2048 bits dhparam generation at container's first launch will take 100% of a core for an extended period of time (10~20 minutes). 

Certificate generation where not super fast compared to `amd64` but way faster than the dhparam generation.

**Comment by buchdag** [comment]
Okay there seems to be an issue indeed, I kept the three containers running on my raspberry pi 3 and forgot about them. I just checked and I saw the exact same thing as you, `letsencrypt_service` is stuck and consume 100% of a CPU core. I have no idea why it get stuck yet.

I'll let it run for a while on an `ARM64-2GB` to check if arm64 is affected too.

Any help with troubleshooting is welcome.

**Comment by buchdag** [comment]
I've had the three containers up and running for 14 hours on an Scaleway ARM64-2GB instance, everything is fine there. The letsencrypt container correctly trigger the service every hour:

```
[...]
Sleep for 3600s
/etc/nginx/certs/my.domain.tld /app
Creating/renewal my.domain.tld certificates... (my.domain.tld)
2018-01-29 07:48:04,434:INFO:simp_le:1534: Certificates already exist and renewal is not necessary, exiting with status code 1.
/app
Sleep for 3600s
[...]
```

And no abnormal CPU utilization either:

![screen shot 2018-01-29 at 08 47 34](https://user-images.githubusercontent.com/13685826/35499543-13d2b2be-04d3-11e8-962b-9818b152b041.png)

So that appears to be an armhf only issue.

**Comment by AKKYM** [comment]
Thank you for your immediate response.
I'm sorry for the late reply.

I did not understand that Scaleway C1 is arm64, so I started this image with armhf settings.
I restarted this image with arm64 setting now.
I will report whether it is working normally after several hours.

**Comment by buchdag** [comment]
Sorry for the misunderstanding, C1 are indeed armhf (armv7l to be exact). Don't waste money with a test on a C1 + arm64 `docker-gen`, it won't work.

I wasn't able to start a new C1 instance, I'll try to troubleshoot this on a Pi (same arch) instead. It seems to be a bash scripting issue on the lestencrypt service, so I hope it won't be too hard to identify and fix.

The arm64 instances I was referring to are these : https://www.scaleway.com/armv8-cloud-servers/ , if those suits your needs It appears to be working ok on them.

**Comment by AKKYM** [comment]
I'm sorry. I misunderstood again.

**Comment by buchdag** [comment]
The culprit was the bash version packaged with alpine 3.7, somehow bash jobs hangs with this version on armhf.

I've opened an issue at Alpine: https://bugs.alpinelinux.org/issues/8447

I've switched the Dockerfiles back to alpine 3.6, that worked on the Raspberry Pi 3, the letsencrypt service isn't stuck anymore.

I was unable to test on a Scaleway C1 because they're out of stock, but that should work as well.
